### PR TITLE
Updated California Institute of Technology (Caltech) Faculty.

### PR DESCRIPTION
**Pull request by Robertboy18**
# Updating all of the Caltech Faculty. 

The current list of faculty at Caltech on csrankings.org hasn't been updated in a while. We have had a bunch of new and established faculty join us so updating all of them. See complete list here: https://www.cms.caltech.edu/people/faculty

# Contributing to CSrankings

Thanks for contributing to CSrankings! Please read and indicate you agree with **all** these guidelines to getting your pull request accepted. Note that pull requests may take some time to get merged (please don't contact us for at least three months for updates, unless you are a sponsor - see below).

**If you find CSrankings useful, [please consider becoming a sponsor](https://github.com/sponsors/csrankings).**

**NOTE:** IF YOU DO NOT FOLLOW THE STEPS BELOW, YOUR PULL REQUEST WILL BE SUMMARILY REJECTED.
You must read and check **all** the boxes below by filling them in with an X or your PR will be rejected.

**The Basics**

- [X] All pull requests and issues must come from non-anonymous accounts. Make sure your GitHub profile contains your full name.

- [X] Use a reasonable title that explains what the PR corresponds to (as in, not "Update csrankings-x.csv").

- [X] Combine multiple updates to a single institution into a **single PR.**

- [X] Only submit one pull request per institution.

- [X] Do not modify any files except `csrankings-[a-z].csv` or (if needed) `country-info.csv` and `old/industry.csv` (see below).

- [X] Do not use Excel to edit any .csv files; Excel incorrectly tries to
convert some Google Scholar entries to formulas, corrupting the
database. Use the GitHub user interface or a text editor like emacs or NotePad instead.

- [X] Insert new faculty **in alphabetical order** (not at the end) in the appropriate `csrankings-[a-z].csv` files. **Do not modify `csrankings.csv`, which is auto-generated.**

- [X] Check to make sure that you have no spaces after commas, or any missing fields.

- [X] Check to make sure the home page is correct.

- [X] Make sure the Google Scholar IDs are just the alphanumeric identifier (not a URL or with `&hl=en`).

- [X] Check to make sure the name corresponds to the DBLP entry (look it up at http://dblp.org).

- [ ] If a faculty member is not in a CS department or similar, include a comment explaining how they meet the inclusion criteria (see below).

**Inclusion criteria**

- [X] Make sure that any faculty you add meet the inclusion
criteria. Eligible faculty include only full-time, tenure-track research
faculty members on a given campus who can *solely* advise PhD students in
Computer Science. Faculty not in a CS department or similar who can
advise PhD students in CS can be included regardless of their home
department. **Provide justification, pointing to specific home pages
showing how faculty not in a CS department meet the inclusion criteria,
e.g. showing a courtesy appointment in CS.** Faculty must also have a 75%+ time appointment (check
`old/industry.csv` for faculty who are now more than 25% in industry).

**Updating an affiliation or home page**

- [ ] Update affiliations, home pages, and Google Scholar entries by modifying `csrankings-[a-z].csv`. For the Google Scholar entry, just use the alphanumeric identifier in the middle of the URL. If none is there, put `NOSCHOLARPAGE`.

**Adding one or more faculty members (including an entire department)**

- [ ] If the department is not yet listed in CSrankings, the entire CS faculty needs to be added (not just one faculty member).

- [ ] Enter each faculty member's [DBLP](http://dblp.org) name, home page, and Google Scholar entry (just the alphanumeric identifier, not the whole URL) by modifying `csrankings-[a-z].csv` (**the letters correspond to the first letter of the faculty members' names**); include disambiguation suffixes like `0001` as needed. If the faculty entry is currently ambiguous, please do not include them. Send mail to the DBLP maintainers (dblp@dagstuhl.de) with a few publications by a particular faculty member; also, open an issue so that when the DBLP database is updated, that faculty member's information can be added.

- [ ] If DBLP has multiple entries for this person, *all of them need to be listed*. Do not update `dblp-aliases.csv`.

- [ ] If the institution you are adding is not in the US,
update `country-info.csv` and add *all* of the faculty in the CS department.

**(Advanced) Quick contribution via a shallow clone** 

We recommend that you use the GitHub web user interface to make changes. However, it may be more convenient
to clone the repository for larger-scale changes.

However, a full clone of the CSrankings repository is almost 2GB.  To
contribute a change without creating a full local clone of the
CSrankings repo, you can perform a _shallow clone_.  To do so, follow these
steps:

1. Fork the CSrankings repo.  If you have an existing fork, but it is not up to date with the main repository, this technique may not work.  If necessary, delete and re-create your fork to get it up to date.  (Do **not** delete your existing fork if it has unmerged changes you want to preserve!)

1. Do a shallow clone of your fork: `git clone --depth 1 https://github.com/yourusername/CSrankings`.  This will only download the most recent commit, not the full git history.

1. Make your changes on a branch, push them to your clone, and create a pull request on GitHub as usual.

If you want to make another contribution and some time has passed, perform these steps again, creating a fresh fork and shallow clone.

### CODE REVIEW

**Comment by Robertboy18** [comment]
It seems like for some of the authors, there is the SSL verification check, even though their website links are correct. Any help on how to fix this? 
>Processing John Doyle,California Institute of Technology,https://doyle.caltech.edu/,C6DtGmMAAAAJ
  Checking https://dblp.org/search/author/api?q=author%3AJohn%20Doyle:$%3A&format=json&c=10
  Checking homepage URL (https://doyle.caltech.edu/)
  ERROR: An exception occurred: HTTPSConnectionPool(host='doyle.caltech.edu', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1007)')))

File: csrankings-b.csv [hunk 1]
```diff
@@ -15,6 +15,7 @@ B. V. Raghavendra Rao,IIT Madras,http://www.cse.iitm.ac.in/~bvrr,jFu4ljYAAAAJ
 Baba C. Vemuri,University of Florida,https://www.cise.ufl.edu/~vemuri,1fEZ_osAAAAJ
 Babak A. Farshchian,NTNU,https://www.ntnu.edu/employees/babak.farshchian,fGgCWqgAAAAJ
 Babak Falsafi,EPFL,http://parsa.epfl.ch/~falsafi,LaNEuBUAAAAJ
+Babak Hassibi,California Institute of Technology,https://www.babak.caltech.edu/,1XoZPhEAAAAJ
```
Comment (emeryberger) [inline csrankings-b.csv:18]: EE not CS

File: csrankings-j.csv [hunk 2]
```diff
@@ -1503,6 +1503,7 @@ Joel Reardon,University of Calgary,http://pages.cpsc.ucalgary.ca/~joel.reardon,N
 Joel S. Bader,Johns Hopkins University,https://www.baderzone.org/,-dyxvHoAAAAJ
 Joel Spencer,New York University,http://cs.nyu.edu/spencer,AmmWYgEAAAAJ
 Joel Voldman,Massachusetts Institute of Technology,http://www.rle.mit.edu/biomicro,NOSCHOLARPAGE
+Joel W. Burdick,California Institute of Technology,https://robotics.caltech.edu/wiki/index.php/Main_Page,NOSCHOLARPAGE
```
Comment (emeryberger) [inline csrankings-j.csv:1506]: Mech E, not CS

File: csrankings-j.csv [hunk 3]
```diff
@@ -1605,6 +1606,7 @@ John Derrick,University of Sheffield,https://www.sheffield.ac.uk/dcs/people/acad
 John Dickerson 0001,University of Maryland - College Park,http://jpdickerson.com,QgDpfCQAAAAJ
 John Dingliana,Trinity College Dublin,https://www.scss.tcd.ie/John.Dingliana,gkc2ZfIAAAAJ
 John Dowell,University College London,http://www0.cs.ucl.ac.uk/people/J.Dowell.html,NOSCHOLARPAGE
+John Doyle,California Institute of Technology,https://doyle.caltech.edu/,C6DtGmMAAAAJ
```
Comment (emeryberger) [inline csrankings-j.csv:1609]: Not CS

File: csrankings-j.csv [hunk 4]
```diff
@@ -1698,6 +1700,7 @@ John Paul Dickerson,University of Maryland - College Park,http://jpdickerson.com
 John Peter Lewis,Victoria University of Wellington,http://ecs.victoria.ac.nz/Main/JohnLewis,wC_k2-4AAAAJ
 John Philip Collomosse,University of Surrey,http://personal.ee.surrey.ac.uk/Personal/J.Collomosse,yVnCUg0AAAAJ
 John Power,University of Bath,http://www.bath.ac.uk/comp-sci/contacts/academics/john_power,mCvZPDn8heUC
+John Preskill,California Institute of Technology,https://www.its.caltech.edu/~preskill/,K2BMBtcAAAAJ
```
Comment (emeryberger) [inline csrankings-j.csv:1703]: Not CS

File: csrankings-r.csv [hunk 5]
```diff
@@ -747,6 +747,7 @@ Richard M. Fujimoto,Georgia Institute of Technology,http://www.cc.gatech.edu/~fu
 Richard M. Jiang 0001,Lancaster University,https://www.lancaster.ac.uk/scc/about-us/people/richard-jiang,NuyoNc4AAAAJ
 Richard M. Karp,Univ. of California - Berkeley,https://www.eecs.berkeley.edu/Faculty/Homepages/karp.html,NOSCHOLARPAGE
 Richard M. Mortier,University of Cambridge,http://mort.io,9LJgRFAAAAAJ
+Richard M. Murray,California Institute of Technology,https://murray.cds.caltech.edu/,NOSCHOLARPAGE
```
Comment (emeryberger) [inline csrankings-r.csv:750]: Not CS

File: csrankings-s.csv [hunk 6]
```diff
@@ -1540,6 +1540,7 @@ Soohyun Nam Liao,Univ. of California - San Diego,https://datascience.ucsd.edu/pe
 Soon Ae Chun,CUNY,http://www.csi.cuny.edu/faculty/school_of_business/CHUN_SOON.html,mE0JBT0AAAAJ
 Soon J. Hyun,KAIST,http://biophotonics.kaist.ac.kr/the-team,P6EuWkgAAAAJ
 Soon Joo Hyun,KAIST,http://biophotonics.kaist.ac.kr/the-team,P6EuWkgAAAAJ
+Soon-Jo Chung,California Institute of Technology,https://aerospacerobotics.caltech.edu/,-ClLU3EAAAAJ
```
Comment (emeryberger) [inline csrankings-s.csv:1543]: Not obviously in CS

**Review by emeryberger — CHANGES_REQUESTED** [review]

**Comment by emeryberger** [comment]
Numerous non-CS faculty with no justification provided.

> Provide justification, pointing to specific home pages
> showing how faculty not in a CS department meet the inclusion criteria,
> e.g. showing a courtesy appointment in CS.

**Pull request closed without merging**